broker/lib/fabrik/DirectorManager.js

```diff
@@ -842,6 +850,44 @@ class DirectorManager extends BaseManager {
       });
   }
 
+  getBackupOperationState20(opts) {
+    const agent_ip = opts.agent_ip;
+    const options = _.assign({
+      service_id: this.service.id,
+      plan_id: this.plan.id,
+      tenant_id: opts.context ? this.getTenantGuid(opts.context) : opts.tenant_id
+    }, opts);
+
+    function isFinished(state) {
+      return _.includes(['succeeded', 'failed', 'aborted'], state);
+    }
+
+    return this.agent
```

Comment:
Ideally, the getbackupState should not go to agent. It should get the info from the apiserver.

The patching of backupstore metadata with logs as well as the update to the eventmesh server shuld happen from the backupmanager via its poller.

Reply:
In that case the implementation should also move out of here I guess.

Reply:
Done